api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java

```diff
@@ -135,22 +140,38 @@ private ConfigurationBeanLoader(LockableModel locking, ServletContext ctx,
 	/**
 	 * Load the instance with this URI, if it is assignable to this class.
 	 */
-	public <T> T loadInstance(String uri, Class<T> resultClass)
+  public <T> T loadInstance(String uri, Class<T> resultClass) throws ConfigurationBeanLoaderException {
+    instancesMap.clear();
+    T result = loadSubordinateInstance(uri, resultClass);
+    instancesMap.clear();
+    return result;
+  }
+	
+	protected <T> T loadSubordinateInstance(String uri, Class<T> resultClass)
 			throws ConfigurationBeanLoaderException {
 		if (uri == null) {
 			throw new NullPointerException("uri may not be null.");
 		}
 		if (resultClass == null) {
 			throw new NullPointerException("resultClass may not be null.");
 		}
-
+		if (instancesMap.containsKey(uri)) {
+		  try {
+		    T t = (T) instancesMap.get(uri);
+        return t;
```

Comment:
Can you please correct layouting of this line

Reply:
Fixed. Thanks.